test/browser/tests/tenant.spec.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright 2021 Opstrace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from "@playwright/test";
+
+import { test } from "../fixtures/authenticated";
+import { logUserIn } from "../utils/authentication";
+
+test.describe("after auth0 authentication", () => {
+  test.beforeEach(logUserIn);
+
+  test.beforeEach(async ({ page }) => {
+    await page.hover("[data-test='sidebar/clusterAdmin/Tenants']");
+    await page.click("[data-test='sidebar/clusterAdmin/Tenants']");
+    expect(await page.isVisible("[data-test='tenant/list']")).toBeTruthy();
+  });
+
+  test("user can create a new Tenant", async ({ page, cluster }) => {
+    const newTenant = `treetops-${Math.floor(Math.random() * 10000)}`;
```

Comment:
That is the name of the tenant, right? In this case I always like to encode that idea in the variable name, e.g. `tenantName`.

Note that you choose a `dash` here in the name of the tenant. 

We still need to add tenant name validation to the API, see https://github.com/opstrace/opstrace/issues/710#issuecomment-869479025 -- would still greatly appreciate help.

When we do this, we will find that (as of today) a dash should not be allowed: https://github.com/opstrace/opstrace/issues/957

Looking forward to seeing these tests here then also _cover_ the API-based tenant name validation.

Reply:
(just ideas that ran through my head while skimming over this -- feel free to ignore for now, whatever makes sense to you)

Reply:
that's a shame about dashing not being allowed, I'll just change to an underscore. Regarding the var name we use `tenant` to mean the name sometimes and sometimes not. Given the ID and the Name are the same thing it's all a bit confusing. :shrug: